### PR TITLE
Add dot graph renderer for new CFG structure

### PR DESCRIPTION
**Pull request by casperboone**
This PR builds on top of #25 , for easy review of this PR [click here](https://github.com/cafejojo/schaapi/pull/26/files/bdecb97d3430b739929d188c5ef23ecdcc720730..7adfe39b4f931e4e960a286ae1b8e532cd069dba).
___
This PR adds a new dot graph renderer for the changed CFG format.

### CODE REVIEW

File: subprojects/usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/usagegraphgenerator/DotGraphRenderer.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        render(cfg)
+        result.append("}")
+
+        return result.toString()
+    }
+
+    private fun render(node: Node) {
+        visited.add(node)
+
+        result.append("    \"${node.hashCode()}\" [shape=ellipse, label=\"${getNodeLabel(node)}\"]\n")
```
Comment (FWDekker) [inline subprojects/usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/usagegraphgenerator/DotGraphRenderer.kt]: I saw those spaces in the original code as well. What are they there for? Is it just part of the DOT specification?

Comment (casperboone) [reply subprojects/usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/usagegraphgenerator/DotGraphRenderer.kt]: They just make the output look pretty, no special reason

Comment (casperboone) [reply subprojects/usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/usagegraphgenerator/DotGraphRenderer.kt]: They just make the output look pretty, no special reason

File: subprojects/usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/usagegraphgenerator/DotGraphRenderer.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        node.successors.forEach { successor ->
+            result.append("    \"${node.hashCode()}\" -> \"${successor.hashCode()}\"\n")
+
+            if (!visited.contains(successor)) render(successor)
+        }
+    }
+
+    private fun getNodeLabel(node: Node) = node.toString().replace("\"", "^")
```
Comment (JSAbrahams) [inline subprojects/usage-graph-generator/src/main/kotlin/org/cafejojo/schaapi/usagegraphgenerator/DotGraphRenderer.kt]: Why not replace `"` with `'`?

**Pull request merged**